numba/core/rvsdg_frontend/bcinterp.py

```diff
@@ -0,0 +1,760 @@
+                    self.current_block.append(br)
+            # Prepare outgoing variables
+            data = {k: self.vsmap[vs] for k, vs in block.out_vars.items()}
+            return data
+        elif isinstance(block, DDGControlVariable):
+            # Emit body
+            with self.set_block(
+                self._get_label(block.name),
+                ir.Block(scope=self.scope, loc=self.loc),
+            ):
+                for cp, v in block.variable_assignment.items():
+                    const = ir.Const(v, loc=self.loc, use_literal_type=False)
+                    self.store(const, f"$.cp.{cp}", redefine=False)
+            return data
+        elif isinstance(block, DDGBranch):
+            # Emit body
+            if len(block.branch_value_table) == 2:
+                self._emit_two_way_switch(block)
+            else:
+                self._emit_n_way_switch(block)
+            return data
+        else:
+            raise NotImplementedError(block.name, type(block))
+
+    def _emit_two_way_switch(self, block):
+        with self.set_block(
+            self._get_label(block.name),
+            ir.Block(scope=self.scope, loc=self.loc),
+        ):
+            # Handle simple two-way branch
+            assert set(block.branch_value_table.keys()) == {0, 1}
+            cp = block.variable
+            cpvar = self.scope.get_exact(f"$.cp.{cp}")
+            truebr = self._get_label(block.branch_value_table[1])
+            falsebr = self._get_label(block.branch_value_table[0])
+            br = ir.Branch(
+                cond=cpvar,
+                truebr=truebr,
+                falsebr=falsebr,
+                loc=self.loc,
+            )
+            self.current_block.append(br)
+
+    def _emit_n_way_switch(self, block):
+        # with self.set_block(
+        #         self._get_label(block.name),
+        #         ir.Block(scope=self.scope, loc=self.loc)):
+        #     # Handle simple two-way branch
+        #     # assert set(block.branch_value_table.keys()) == {0, 1}
+        #     cp = block.variable
+        #     cpvar = self.scope.get_exact(f"$.cp.{cp}")
+
+        #     falsebr = self._get_label(block.branch_value_table[1])
+        #     truebr = self._get_label(block.branch_value_table[0])
+
+        #     const = self.store(ir.Const(0, loc=self.loc), "$.const")
+        #     cmp = ir.Expr.binop(operator.eq, const, cpvar,
+        #                         loc=self.loc)
+        #     pred = self.store(cmp, "$.pred")
+        #     br = ir.Branch(
+        #         cond=pred,
+        #         truebr=truebr,
+        #         falsebr=falsebr,
+        #         loc=self.loc,
+        #     )
+        #     self.current_block.append(br)
+        bvt = block.branch_value_table
+        cp = block.variable
+        cpvar = self.scope.get_exact(f"$.cp.{cp}")
+        labels = [(k, self._get_label(v)) for k, v in bvt.items()]
+
+        blocks = []
+        for _ in labels[:-1]:
```

Comment:
```suggestion
        for _ in len(labels) - 1:
```

To improve readability.

Reply:
done in aeef2424e697d006dab8b1e09ea3627157b5199c